src/lava/proc/io/bridge/injector.py

```diff
@@ -0,0 +1,176 @@
+
+    For the receiving part of the injector_channel (recv_port), the following
+    properties can be parametrized:
+        (1) When the channel buffer is empty, receiving will either:
+            (a) Block until an item is available (until the sending part
+            sends an item).
+            (b) Not block, and the received data will be zeros.
+        (2) When the channel buffer is not empty, receiving will either:
+            (a) Receive a single item, being the oldest item put in the
+            channel (FIFO).
+            (b) Receive all items available, accumulated.
+
+    Parameters
+    ----------
+    shape : tuple
+        Shape of the OutPort of the Process.
+    buffer_size : optional, int
+        Buffer size of the injector_channel.
+    injector_channel_config : optional, ChannelConfig
+        Configuration object specifying how the src_port behaves when the
+        buffer is full and how the dst_port behaves when the buffer is empty
+        and not empty.
+    """
+    def __init__(self,
+                 shape: tuple[int, ...],
+                 buffer_size: ty.Optional[int] = 50,
+                 injector_channel_config: ty.Optional[ChannelConfig] =
+                 ChannelConfig()) \
+            -> None:
+        super().__init__()
+
+        validate_shape(shape)
+        validate_buffer_size(buffer_size)
+        validate_channel_config(injector_channel_config)
+
+        self._injector_channel_config = injector_channel_config
+
+        self._multi_processing = MultiProcessing()
+        self._multi_processing.start()
+
+        self._injector_channel = \
+            PyPyChannel(message_infrastructure=self._multi_processing,
+                        src_name="src",
+                        dst_name="dst",
+                        shape=shape,
+                        dtype=float,
+                        size=buffer_size)
+        self._injector_channel_src_port = self._injector_channel.src_port
+        self._injector_channel_src_port.start()
+
+        self.proc_params["shape"] = shape
+        self.proc_params["injector_channel_config"] = injector_channel_config
+        self.proc_params["injector_channel_dst_port"] = \
+            self._injector_channel.dst_port
+
+        if self._injector_channel_config.send_buffer_full == \
+                ChannelSendBufferFull.BLOCKING:
+            self._send_data = send_data_blocking
+        elif self._injector_channel_config.send_buffer_full == \
+                ChannelSendBufferFull.NON_BLOCKING_DROP:
+            self._send_data = send_data_non_blocking_drop
+
+        self.out_port = OutPort(shape=shape)
+
+    def send_data(self, data: np.ndarray) -> None:
+        """Send data to the ProcessModel.
+
+        The data is sent through injector_channel.src_port.
+
+        Parameters
+        ----------
+        data : np.ndarray
+            Data to be sent.
+        """
+        validate_send_data(data, self.out_port)
```

Comment:
For performance critical code, it's best not to put any manual type checking / parameter assertions in the "tight loop".

In this case, it's better to let the incorrect data go through and generate a probably worse error message later, than to pay a performance cost on every single timestep of a correctly implemented model. This is especially true here because static type checking can be used in any modern IDE to let people know they're passing the wrong kind of thing.